src/components/MAPI/workernodes/WorkerNodesCreateNodePoolAvailabilityZones.tsx

```diff
@@ -0,0 +1,184 @@
+import { getProvider } from 'stores/main/selectors';
+import useSWR from 'swr';
+import InputGroup from 'UI/Inputs/InputGroup';
+
+import {
+  INodePoolPropertyProps,
+  withNodePoolAvailabilityZones,
+} from './patches';
+
+interface IWorkerNodesCreateNodePoolAvailabilityZonesProps
+  extends INodePoolPropertyProps,
+    Omit<React.ComponentPropsWithoutRef<typeof InputGroup>, 'onChange' | 'id'> {
+  cluster: Cluster;
+}
+
+const WorkerNodesCreateNodePoolAvailabilityZones: React.FC<IWorkerNodesCreateNodePoolAvailabilityZonesProps> = ({
+  id,
+  nodePool,
+  onChange,
+  readOnly,
+  disabled,
+  autoFocus,
+  cluster,
+  ...props
+}) => {
+  const clientFactory = useHttpClientFactory();
+  const auth = useAuthProvider();
+
+  const controlPlaneNodeListKey = cluster
+    ? fetchMasterListForClusterKey(cluster)
+    : null;
+
+  const {
+    data: controlPlaneNodeList,
+    error: controlPlaneNodeListError,
+  } = useSWR<ControlPlaneNodeList, GenericResponseError>(
+    controlPlaneNodeListKey,
+    () => fetchMasterListForCluster(clientFactory, auth, cluster)
+  );
+
+  useEffect(() => {
+    if (controlPlaneNodeListError) {
+      ErrorReporter.getInstance().notify(controlPlaneNodeListError);
+    }
+  }, [controlPlaneNodeListError]);
+
+  const controlPlaneZones = useMemo(() => {
+    if (
+      typeof controlPlaneNodeListError !== 'undefined' ||
+      !controlPlaneNodeList
+    ) {
+      return undefined;
+    }
+
+    return computeControlPlaneNodesStats(controlPlaneNodeList.items)
+      .availabilityZones;
+  }, [controlPlaneNodeList, controlPlaneNodeListError]);
+
+  const provider = useSelector(getProvider);
+  const azStats = useSelector(getSupportedAvailabilityZones);
+
+  const [azSelector, setAzSelector] = useState(
+    AvailabilityZoneSelection.Automatic
+  );
+  const [autoZonesCount, setAutoZonesCount] = useState(1);
+  const [autoZoneIsValid, setAutoZoneIsValid] = useState(true);
+  const [manualZones, setManualZones] = useState<string[]>([]);
+  const [manualZonesIsValid, setManualZonesIsValid] = useState(false);
+
+  const handleChange = (selector: AvailabilityZoneSelection) => {
+    setAzSelector(selector);
+  };
+
+  useEffect(() => {
+    if (!controlPlaneZones) return;
+
+    switch (azSelector) {
+      case AvailabilityZoneSelection.Automatic:
+        onChange({
+          isValid: autoZoneIsValid,
+          patch: withNodePoolAvailabilityZones(
+            determineRandomAZs(autoZonesCount, azStats.all, controlPlaneZones)
+          ),
+        });
+        break;
+
+      case AvailabilityZoneSelection.Manual:
+        onChange({
+          isValid: manualZonesIsValid,
+          patch: withNodePoolAvailabilityZones(manualZones),
+        });
+        break;
+
+      case AvailabilityZoneSelection.NotSpecified:
+        onChange({
+          isValid: true,
+          patch: withNodePoolAvailabilityZones(undefined),
+        });
+        break;
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Disabling the rule because it requires that we add `onChange` to the list of dependencies, but that results in a memory leak, since `onChange` has a different pointer in every re-render.